uaclient/system.py

```diff
@@ -0,0 +1,213 @@
+import os
+import re
+import uuid
+from functools import lru_cache
+from typing import Dict, Optional, Set
+
+from uaclient import exceptions, util
+
+REBOOT_FILE_CHECK_PATH = "/var/run/reboot-required"
+REBOOT_PKGS_FILE_PATH = "/var/run/reboot-required.pkgs"
+ETC_MACHINE_ID = "/etc/machine-id"
+DBUS_MACHINE_ID = "/var/lib/dbus/machine-id"
+
+# N.B. this relies on the version normalisation we perform in get_platform_info
+REGEX_OS_RELEASE_VERSION = r"(?P<release>\d+\.\d+) (LTS )?\((?P<series>\w+).*"
+
+
+@lru_cache(maxsize=None)
+def get_machine_id(cfg) -> str:
+    """Get system's unique machine-id or create our own in data_dir."""
+    # Generate, cache our own uuid if not present in config or on the system
```

Comment:
I think the phrasing of the second sentence is confusing. I prefer the one we already have in the docstring.  We can improve the docstring by just adding that we can retrieve the `machine_id` from the `machine_token` file.

Reply:
Good point! I only copy pasted this, but I added another commit here to make the adjustments you suggest.